state-chain/chains/src/sol.rs

```diff
@@ -219,17 +212,30 @@ impl FeeEstimationApi<Solana> for SolTrackedData {
 				},
 		);
 
-		LAMPORTS_PER_SIGNATURE.saturating_add(
-			// It should never approach overflow but just in case
-			sp_std::cmp::min(
-				SolAmount::MAX as u128,
-				(self.priority_fee as u128 * compute_units_per_fetch as u128)
-					.div_ceil(MICROLAMPORTS_PER_LAMPORT.into()),
-			) as SolAmount,
-		)
+		calculate_gas_fee(self.priority_fee, compute_units_per_fetch)
 	}
 }
 
+// Calculate the gas fee for a transaction given its compute units and a priority fee.
+fn calculate_gas_fee(
```

Comment:
Could impl this on `SolTrackedData` and then we don't need to pass in self.priority_fee